source_handlers.go

```diff
@@ -325,3 +325,30 @@ func SourceCheckAvailability(c echo.Context) error {
 
 	return c.JSON(http.StatusAccepted, map[string]interface{}{})
 }
+
+func InternalSourceList(c echo.Context) error {
```

Comment:
Might I suggest moving this handlerFunc to a different file? Maybe could have an `internal_handlers.go` file - that way we can separate out the 2 internal endpoints there.  

What do you think?

Reply:
I like the idea. Applying it now.